core/src/workflow/workflow_tasks/cache_manager.rs

```diff
@@ -55,8 +105,19 @@ impl WorkflowCacheManager {
     }
 
     pub fn remove(&mut self, run_id: &str) {
+        let outstanding = self.outstanding_eviction.take();
+        if matches!(outstanding.as_ref(), Some(r) if r != run_id) {
+            // TODO: This isn't an error since eviction could be completed out-of-order?
+            error!("Evicted not-outstanding run");
```

Comment:
This likely won't be an error, but I need to add some tests around this. It's unclear to me exactly how this can happen, but it can.